lib/modules/platform/gitlab/index.ts

```diff
@@ -1005,21 +1008,34 @@ export async function setBranchStatus({
     options.target_url = targetUrl;
   }
 
-  if (branchSha) {
-    const commitUrl = `projects/${config.repository}/repository/commits/${branchSha}`;
-    await gitlabApi
-      .getJsonSafe(commitUrl, LastPipelineId)
-      .onValue((pipelineId) => {
-        options.pipeline_id = pipelineId;
-      });
-  }
+  const retryTimes = parseInteger(
+    process.env.RENOVATE_X_GITLAB_BRANCH_STATUS_CHECK_ATTEMPTS,
+    2,
+  );
 
   try {
-    // give gitlab some time to create pipelines for the sha
-    await setTimeout(
-      parseInteger(process.env.RENOVATE_X_GITLAB_BRANCH_STATUS_DELAY, 1000),
-    );
+    for (let attempt = 1; attempt <= retryTimes; attempt += 1) {
```

Comment:
Is this intended to skip the for loop if the value is unset (value 0)?

It's kind of a confusing way to do this, I'd prefer it is more explicit/obvious for future maintainability